### PR TITLE
enhancement(datadog_agent source): enable zstd decompression

### PR DESCRIPTION
**Pull request by StephenWakely**
The datadog agent is in the process of allowing payloads to be compressed with `zstd`. This adds the ability to Vector to handle these.

### CODE REVIEW

File: changelog.d/20732_datadog_agent_zstd.enhancement.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Allow the datadog agent source to accept payloads that have been compressed with `zstd`.
```
Comment (jszwedko) [inline changelog.d/20732_datadog_agent_zstd.enhancement.md:1]:
```suggestion
Allow the `datadog_agent` source to accept payloads that have been compressed with `zstd`.
```

I think we've standardized on either the component name, in backticks, or using capitalization like "Datadog Agent source".

File: scripts/integration/datadog-agent/test.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ env:
   TEST_DATADOG_API_KEY:
 
 matrix:
-  version: ['7']
+  version: ['7.54']
```
Comment (jszwedko) [inline scripts/integration/datadog-agent/test.yaml:14]: Did this turn out to be necessary?

**Review by jszwedko — COMMENTED** [review]
A couple of nits/questions but otherwise 👍

File: scripts/integration/datadog-agent/test.yaml (hunk 2 above)
Comment (StephenWakely) [reply scripts/integration/datadog-agent/test.yaml]: Actually looks like `7` will resolve to 7.55 which should be fine.

**Pull request merged**